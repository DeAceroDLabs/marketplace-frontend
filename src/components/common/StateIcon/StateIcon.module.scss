@use "styles/colors";
.outer-circle {
    display: flex;
    align-items: center;
    width: 128px;
    height: 128px;
    border-radius: 64px;
}

.inner-circle {
    display: flex;
    align-items: center;
    margin: auto;
    width: 104px;
    height: 104px;
    border-radius: 52px;
    svg {
        color: colors.$color-background-01;
        font-size: 50px;
        margin: auto;
    }
}

.success {
    background-color: colors.$color-success-disable;
}

.success-active {
    background-color: colors.$color-success-active;
}

.error {
    background-color: colors.$color-error-disable;
}

.error-active {
    background-color: colors.$color-error-active;
}