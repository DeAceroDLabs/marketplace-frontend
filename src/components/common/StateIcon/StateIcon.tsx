import ErrorOutlineIcon from "@mui/icons-material/ErrorOutline";
import CheckCircleOutlineIcon from "@mui/icons-material/CheckCircleOutline";
import styles from "./StateIcon.module.scss";

interface StateIconProps {
  state: "success" | "error";
}

const StateIcon: React.FC<StateIconProps> = ({ state }) => {
  const icons = {
    error: <ErrorOutlineIcon />,
    success: <CheckCircleOutlineIcon />,
  };

  return (
    <div className={`${styles["outer-circle"]} ${styles[state]}`}>
      <div className={`${styles["inner-circle"]} ${styles[`${state}-active`]}`}>
        {icons[state]}
      </div>
    </div>
  );
};

export default StateIcon;
