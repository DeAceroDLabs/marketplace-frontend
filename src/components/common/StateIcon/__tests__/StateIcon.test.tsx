import { render } from "@testing-library/react";
import StateIcon from "../StateIcon";

describe("StateIcon", () => {
  it("loads without exploding", () => {
    const view = render(<StateIcon state="success" />);
    expect(view).toMatchSnapshot();
  });
});
