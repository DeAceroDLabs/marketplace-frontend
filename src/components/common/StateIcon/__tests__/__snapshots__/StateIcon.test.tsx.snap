// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateIcon loads without exploding 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="outer-circle success"
      >
        <div
          class="inner-circle success-active"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckCircleOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="outer-circle success"
    >
      <div
        class="inner-circle success-active"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckCircleOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
